source/common/rate_limiter_impl.cc

```diff
@@ -78,6 +78,43 @@ void LinearRateLimiter::releaseOne() {
   acquired_count_--;
 }
 
+LinearRampingRateLimiterImpl::LinearRampingRateLimiterImpl(Envoy::TimeSource& time_source,
+                                                           const std::chrono::nanoseconds ramp_time,
+                                                           const Frequency frequency)
+    : RateLimiterBaseImpl(time_source), ramp_time_(ramp_time), frequency_(frequency) {
+  if (frequency_.value() <= 0) {
+    throw NighthawkException("frequency must be > 0");
+  }
+  if (ramp_time <= 0ns) {
+    throw NighthawkException("duration must be positive");
+  }
+}
+
+bool LinearRampingRateLimiterImpl::tryAcquireOne() {
+  if (acquireable_count_) {
+    acquired_count_++;
+    return acquireable_count_--;
+  }
+  const auto elapsed_time = elapsed() + 1ns;
```

Comment:
(optional) This may be a good place to spell out the type instead of using auto to help the reader:

```cc
const std::chrono::nanoseconds elapsed_time = ...
```

Reply:
Done in 6329377a31b0b35828d81c2e2a22cfd8f3943e8a